dbms/src/Storages/DeltaMerge/RowKeyRange.cpp

```diff
@@ -46,6 +46,59 @@ const RowKeyValue RowKeyValue::COMMON_HANDLE_MAX_KEY
     = RowKeyValue(true, std::make_shared<String>(1, TiDB::CodecFlag::CodecFlagMax), 0);
 const RowKeyValue RowKeyValue::EMPTY_STRING_KEY = RowKeyValue(true, std::make_shared<String>(""), 0);
 
+std::pair<RowKeyValue, std::string_view> RowKeyValue::fromHandleWithSuffix(
+    bool is_common_handle_,
+    const HandleValuePtr value_)
+{
+    if (is_common_handle_)
+    {
+        RowKeyValue rowkey_value(is_common_handle_, value_, 0);
+        return {rowkey_value, std::string_view{nullptr, 0}};
+    }
+
+    assert(!is_common_handle_);
+    // According to the tidb encoding rule, the int handle must be larger than 8 bytes.
+    RUNTIME_CHECK_MSG(
```

Comment:
So we allow an encoded handle longer than sizeof(int64) here, but will still panic if we meet an encoded handle shorter than sizeof(int64)

Reply:
Yes, according to @tangenta this case should not happen. If it happens, then tidb-server would panic.

Reply:
And if the value_size is less than 8 bytes, then the `DB::DecodeInt64` will likely access random address, which will cause random panic.
Adding a check here so that we can get explicit error rather than random panic.

Reply:
I've try to generate a split key like
* ` t{tableID}_r\x30`
* ` t{tableID}_r\x00\x00\x00\x00\x00\x00\x01`

TiKV will refuse to split, with logging like
```
[2025/05/09 17:41:36.242 +08:00] [INFO] [peer.rs:6460] ["invalid split request"] [source=pd] [peer_id=2078] [region_id=2077] [err="KeyNotInRegion([116, 128, 0, 0, 0, 0, 0, 1, 255, 255, 215, 95, 114, 0, 0, 0, 0, 255, 0, 250, 95, 114, 0, 0, 0, 0, 255, 0, 0, 1, 0, 0, 0, 0, 0, 250], id: 2077 start_key: 7480000000000001FFD75F730000000000FA end_key: 7480000000000001FFD800000000000000F8 region_epoch { conf_ver: 26 version: 217 } peers { id: 2078 store_id: 1 } peers { id: 2166 store_id: 132 role: Learner })"] [thread_id=377]
```